### DISCUSSION TITLE
How to get the time elapsed inside a step

### DISCUSSION THREAD

**Discussion by Ananda0488** [getgauge/taiko#2356] [General]
My requirement is: Inside a test step, I want to check how much time elapsed. Depending upon the time, I will do some action. Is there a way to get that information?

**Comment by zabil** [comment]
Have you tried using [`waitFor`](https://docs.taiko.dev/api/waitfor/) to wait for a specific amount of time before performing an action?

**Reply by Ananda0488** [reply]
Hi Zabil, thank you for the response. My requirement is not to wait for a specific amount of time and then perform the action. I want to check the time inside a step and once I find that, for example 10 min has elapsed, then I will perform an action.

For example, I was thinking to implement my requirement this way:
In the Gauge report, we can see Execution Time for each step. What I think is, when a step starts executing, a timer is started somewhere and when the step completes execution, the timer is stopped. Is there a way to get the timer value inside the step? If I get the timer value and find that 10 min has elapsed, then I will perform an action.

**Reply by Ananda0488** [reply]
Hi, can anyone please help me regarding this.

**Reply by zabil** [reply]
@Ananda0488 I don't think there's no easy way to do this at the moment. You could try using [data stores](https://docs.gauge.org/writing-specifications.html?os=macos&language=javascript&ide=vscode) and [execution hooks](https://docs.gauge.org/writing-specifications.html?os=macos&language=javascript&ide=vscode#execution-hooks) maybe.

**Reply by Ananda0488** [reply]
ok... will try... thanks Zabil... :)

**Reply by sriv** [reply]
You are right in guessing that Gauge starts a timer, however you should consider that to be the internals of gauge, which includes actions before and after the step code. For your purpose, I would suggest that you use a timer within the step implementation that you have complete control of.

**Reply by Ananda0488** [reply]
ok... thanks Sriv :)
